internal/store/factory.go

```diff
@@ -16,18 +16,42 @@
 package store
 
 import (
+	"fmt"
 	"github.com/edgexfoundry/app-functions-sdk-go/v2/internal/store/db"
-	"github.com/edgexfoundry/app-functions-sdk-go/v2/internal/store/db/interfaces"
 	"github.com/edgexfoundry/app-functions-sdk-go/v2/internal/store/db/redis"
-
+	"github.com/edgexfoundry/app-functions-sdk-go/v2/pkg/interfaces"
 	bootstrapConfig "github.com/edgexfoundry/go-mod-bootstrap/v2/config"
 )
 
-func NewStoreClient(config db.DatabaseInfo, credentials bootstrapConfig.Credentials) (interfaces.StoreClient, error) {
+type ClientFactory interface {
+	NewStoreClient(config interfaces.DatabaseInfo, credentials bootstrapConfig.Credentials) (interfaces.StoreClient, error)
+	RegisterCustomStoreFactory(name string, factory func(config interfaces.DatabaseInfo, credentials bootstrapConfig.Credentials) (interfaces.StoreClient, error)) error
```

Comment:
Why is `RegisterCustomStoreFactory` part of this interface? Should only need it to be on the `ApplicationService` interface. I think there may be one too many layers here.

Reply:
The factory is stored in the dic so that it's available on config updates where if I remember right the whole service was not in scope.  So the method on the service essentially proxies into this one (thru the dic) to register the custom functions.

Reply:
Getting rid of the bootstrap handler made it more obvious how to get rid of the extra layer :)